ReactiveCocoa/Swift/Property.swift

```diff
@@ -484,30 +484,19 @@ public final class Property<Value>: PropertyProtocol {
 		// Share a replayed producer with `self.producer` and `self.signal` so
 		// they see a consistent view of the `self.value`.
 		// https://github.com/ReactiveCocoa/ReactiveCocoa/pull/3042
-		let producer = unsafeProducer.replayLazily(upTo: 1)
-		
-		// Verify that an initial is sent. This is friendlier than deadlocking
-		// in the event that one isn't.
-		var value: Value? = nil
-		let disposable = producer.start { event in
-			switch event {
-			case let .next(newValue):
-				value = newValue
-				
-			case .completed, .interrupted:
-				break
-				
-			case let .failed(error):
-				fatalError("Receive unexpected error from a producer of `NoError` type: \(error)")
-			}
-		}
-		guard value != nil else {
+		let (producer, lense) = unsafeProducer.replay(upTo: 1)
+
+		// Start the replay producer.
+		producer.start().dispose()
+
+		// Verify that an initial is sent. This is friendlier than deadlocking in
+		// the event that one isn't.
+		guard !lense().isEmpty else {
 			fatalError("A producer promised to send at least one value. Received none.")
 		}
-		disposable.dispose()
 
 		self.sources = sources
-		_value = { producer.take(first: 1).single()!.value! }
+		_value = { lense()[0] }
```

Comment:
Yes, but there's a reason for each suggestion:
- `first` is more explicit in 2 ways:
  - it avoids the magic number `0`.
  - it makes it explicit that the collection might be empty.
- `_value = lense`: this avoids an extra level of indirection.